x-pack/plugins/cloud/server/utils.test.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { parseDeploymentIdFromDeploymentUrl } from './utils';
+
+describe('parseDeploymentIdFromDeploymentUrl', () => {
+  it('should return undefined if there is no deploymentUrl configured', () => {
+    expect(parseDeploymentIdFromDeploymentUrl()).toBeUndefined();
+  });
+
+  it('should return the deploymentId if this is a valid deployment url', () => {
+    expect(parseDeploymentIdFromDeploymentUrl('deployments/uuid-deployment-1')).toBe(
+      'uuid-deployment-1'
+    );
+  });
```

Comment:
I sure hope the `deploymentUrl` format will never change.

Imho it would have been better to have `deploymentId` in the config and to forge `deploymentUrl` from it than the other way around (or even just have both in the config)

cc @yspotts

Reply:
@AlexP-Elastic @andrew-moldovan Deployment id would have been nice ;-)

Reply:
Yes deployment Id will be nice, I put this in the cloud plugin, hoping that in the future this could come from the config directly and we can benefits from that directly in the fleet plugin.

Reply:
See https://github.com/elastic/kibana/pull/98070#pullrequestreview-643286479 below

Fwiw if I was going to stick with deploymentUrl, I'd probably do something like (apologies for pseudocode)`/deployments/([^/]+) || [0-9a-f]{30,}` to increase the chance that some minor change happens in the future and breaks all the running 7.13.0s